asllib/doc/ASLTypingReference.tex

```diff
@@ -538,48 +753,81 @@ \subsection{Prose}
 
 \begin{emptyformal}
     \subsection{Formally}
+
+\[
+\begin{array}{c}
+\inferrule{}{E \entails \unconstrainedinteger : \isunconstrainedinteger}\\
+\\
+\inferrule{}{E \entails \TInt(\langle [] \rangle) : \isunderconstrainedinteger}\\
+\\
+\RuleComment{This is not entirely correct, since the constraints may be}\\
+\RuleComment{malformed, or include under-constrained integers.}\\
+\RuleComment{But this is what I go with, for now.}\\
+\inferrule{}{E \entails \TInt(\langle [c_1,\ldots,c_k]\rangle) : \iswellconstrainedinteger}
+\end{array}
+\]     
+    
+\todocomment{
+ROMAN: see note about domains.
+}
 \end{emptyformal}
 
 \subsection{Comments}
     This aims to encompass LRM Section 3.4.1 D\_ZTPP, R\_WJYH, R\_HJPN, R\_CZTX, R\_TPHR
 
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 \chapter{Type Satisfaction and Related Notions}
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+\todocomment{
+It is not obvious to me why sometimes we check for type satisfaction and other times we check the structure of a type.
+}
 
 \section{TypingRule.Subtype} 
 
   \subsection{Prose}
-    The subtype relation is a partial order.
+The subtype relation is a partial order over named types.
+\todocomment{
+ROMAN: I think the syntax allows subtyping anonymous types as well. Perhaps we should fix the syntax?
+}
```

Comment:
yes please make a proposal as part of this PR

Reply:
Our code actually does the right thing. It's the LRM syntax rule that's wrong.
I think we should add the concrete syntax to the abstract syntax reference document and fix the rule there. I will add it to the list of tasks for the next PR.